### PR TITLE
Hive: Fix RetryingMetaStoreClient for Hive 2.1

### PR DESCRIPTION
**Pull request by szehon-ho**
Fixes : https://github.com/apache/iceberg/issues/3363, RetryingMetaStoreClient broken for Hive 2.1.

The root cause is this commit: https://issues.apache.org/jira/browse/HIVE-12918, fixed in Hive 2.3 (the version we build against) by https://issues.apache.org/jira/browse/HIVE-15081

Change to use this more stable method in RetryingMetaStoreClient: https://github.com/apache/hive/blob/rel/release-2.3.8/metastore/src/java/org/apache/hadoop/hive/metastore/RetryingMetaStoreClient.java#L95.  

I'm usually hesitant to use @VisibleForTesting methods, but as @sunchao pointed out offline to me, for Hive it means the method is at least tested :), unlike the non marked one.

### CODE REVIEW

**Comment by szehon-ho** [comment]
Looks like it passes for at least the tested Hive 1 and 2 versions in Iceberg test environment.  @Reo-LEI do you want to try this one?

**Comment by Reo-LEI** [comment]
Thanks @szehon-ho , I will try this in today or tomorrow.

**Comment by rdblue** [comment]
@szehon-ho, is this good to go?

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java [hunk 1]
```diff
@@ -53,7 +52,7 @@ public HiveClientPool(int poolSize, Configuration conf) {
   protected IMetaStoreClient newClient()  {
     try {
       try {
-        return GET_CLIENT.invoke(hiveConf, true);
+        return GET_CLIENT.invoke(hiveConf, (HiveMetaHookLoader) tbl -> null, HiveMetaStoreClient.class.getName());
```
Comment (Reo-LEI) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java:55]: I test this in hive 2.1.1 and that is work, I think this approach is ok.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @szehon-ho for fixing this. And thanks @Reo-LEI for testing!

**Comment by szehon-ho** [comment]
Yea thanks a lot @Reo-LEI for reporting and verifying and @rdblue for review!

**Pull request merged**